arduinolux/src/Debug/DebugTest.cpp

```diff
@@ -0,0 +1,14 @@
+#include"../StringConverter.h"
+#include <string>
+#include <iostream>
+
+int main() {
+	std::string testString = "Hello there world!";
+	do {
+		char* convertedString = StringConverter::ExtractCharArrayFromString(testString);
+
+		std::cout << convertedString << "\n";
```

Comment:
So, I just tried a version where I replaced this line with:

std::cout << testString.c_str() << "\n";

and it behaved as I expected. What did your c_str() version look like?

I'm happy to merge pulls since I'm not really maintaining this these days, or to just add you as a collaborator on the rogue-hack-lab/translux repo so you can choose when to merge pulls from your fork when you want.